### PR TITLE
#4046 Enable Shabondi JMX port and HEAP_OPTS

### PR DESCRIPTION
**Pull request by vitojeng**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #4046)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-shabondi/src/main/scala/oharastream/ohara/shabondi/ShabondiDefinitions.scala [hunk 1]
```diff
@@ -137,6 +147,24 @@ object ShabondiDefinitions {
...
+    .optional(1024L)
+    .documentation("maximum memory allocation (in MB)")
+    .build
+    .registerTo(basicDefinitionMap)
+
+  val INIT_HEAP_DEFINITION = SettingDef.builder
+    .group(CORE_GROUP)
+    .key("xms")
+    .orderInGroup(orderInGroup())
+    .optional(1024L)
```
Comment (chia7712) [inline ohara-shabondi/src/main/scala/oharastream/ohara/shabondi/ShabondiDefinitions.scala:163]: ```positiveNumber```

File: ohara-shabondi/src/main/scala/oharastream/ohara/shabondi/ShabondiDefinitions.scala [hunk 2]
```diff
@@ -137,6 +147,24 @@ object ShabondiDefinitions {
     .build
     .registerTo(basicDefinitionMap)
 
+  val MAX_HEAP_DEFINITION = SettingDef.builder
+    .group(CORE_GROUP)
+    .key("xmx")
+    .orderInGroup(orderInGroup())
+    .optional(1024L)
```
Comment (chia7712) [inline ohara-shabondi/src/main/scala/oharastream/ohara/shabondi/ShabondiDefinitions.scala:154]: ```positiveNumber```

File: ohara-it/src/test/scala/oharastream/ohara/it/shabondi/TestShabondi.scala [hunk 3]
```diff
@@ -283,6 +246,8 @@ class TestShabondi(platform: ContainerPlatform) extends WithRemoteConfigurator(p
       resultInfo.deadNodes shouldBe Set.empty
     }
 
+    testJmx(clusterInfo)
```
Comment (vitojeng) [inline ohara-it/src/test/scala/oharastream/ohara/it/shabondi/TestShabondi.scala:249]: @chia7712 確認一下...
在 Shabondi IT 的 jmx 測試, 就是只有在 shabondi start 後, 呼叫這個方法測試而已.
目前似乎看來沒有別的部份.

Comment (chia7712) [reply ohara-it/src/test/scala/oharastream/ohara/it/shabondi/TestShabondi.scala:249]: 進階一點就是打一些資料進去 看一下有沒有特定的metrics產生

不過我是覺得現階段這樣就可以了

File: ohara-shabondi/src/main/scala/oharastream/ohara/shabondi/ShabondiDefinitions.scala (hunk 1 above)
Comment (vitojeng) [reply ohara-shabondi/src/main/scala/oharastream/ohara/shabondi/ShabondiDefinitions.scala]: done

File: ohara-shabondi/src/main/scala/oharastream/ohara/shabondi/ShabondiDefinitions.scala (hunk 2 above)
Comment (vitojeng) [reply ohara-shabondi/src/main/scala/oharastream/ohara/shabondi/ShabondiDefinitions.scala]: done

File: ohara-client/src/main/scala/oharastream/ohara/client/configurator/v0/ShabondiApi.scala [hunk 4]
```diff
@@ -125,6 +126,10 @@ final object ShabondiApi {
     def clientPort(port: Int): Request.this.type =
       setting(CLIENT_PORT_DEFINITION.key, JsNumber(CommonUtils.requireBindPort(port)))
 
+    @Optional("the default port is random")
+    def jmxPort(jmxPort: Int): Request.this.type =
+      setting(JMX_PORT_DEFINITION.key(), JsNumber(CommonUtils.requireConnectionPort(jmxPort)))
```
Comment (chia7712) [inline ohara-client/src/main/scala/oharastream/ohara/client/configurator/v0/ShabondiApi.scala:131]: ```requireBindPort```

Comment (chia7712) [reply ohara-client/src/main/scala/oharastream/ohara/client/configurator/v0/ShabondiApi.scala:131]: 不過理論上這個值已經被JsonRefiner檢查過了 這時候應該是可以跳過檢查

Comment (vitojeng) [reply ohara-client/src/main/scala/oharastream/ohara/client/configurator/v0/ShabondiApi.scala]: done

**Comment by vitojeng** [comment]
retry

**Comment by vitojeng** [comment]
retry collieIT

**Comment by vitojeng** [comment]
retry otherIT

**Pull request merged**